app/src/features/surveys/components/SurveyStratumForm.tsx

```diff
@@ -0,0 +1,313 @@
+import { mdiDotsVertical, mdiPencilOutline, mdiPlus, mdiTrashCanOutline } from '@mdi/js';
+import Icon from '@mdi/react';
+import Button from '@mui/material/Button';
+import Typography from '@mui/material/Typography';
+import IconButton from '@mui/material/IconButton';
+import Card from '@mui/material/Card';
+import DialogActions from '@mui/material/DialogActions';
+import {  Formik, FormikProps, useFormikContext } from 'formik';
+import { IEditSurveyRequest } from 'interfaces/useSurveyApi.interface';
+import yup from 'utils/YupSchema';
+import { useState, useRef } from 'react';
+import { Box, Dialog, DialogContent, DialogContentText, DialogTitle, ListItemIcon, Menu, MenuItem, MenuProps, useMediaQuery, useTheme } from '@mui/material';
+import CustomTextField from 'components/fields/CustomTextField';
+
+interface IStratum {
+  /**
+   * @TODO We probably won't track the survey stratum ID in the frontend. Reason: new records won't have an ID.
+   * From a technical perspective, it is much easier to simply erase all stratum and re-insert them upon
+   * survey create/survey update, rather than track the IDs of already existing stratums.
+   */
+
+  // survey_stratum_id: number | undefined;
+  name: string;
+  description: string;
+}
+
+interface IStratumForm {
+  index: number | null;
+  stratum: IStratum;
+}
+
+const StratumFormInitialValues: IStratumForm = {
+  index: null,
+  stratum: {
+    name: '',
+    description: ''
+  }
+}
+
+export const StratumFormYupSchema = yup.object().shape({
+  index: yup.number().nullable(true),
+  stratum: yup.object().shape({
+    survey_stratum_id: yup.number(),
+    name: yup
+      .string()
+      .required('Must provide a Stratum name')
+      .max(300, 'Name cannot exceed 300 characters'),
+    description: yup
+      .string()
+      .required('Must provide a Stratum description')
+      .max(3000, 'Description cannot exceed 3000 characters'),
+  })
+});
+
+interface IStratumDialogProps {
+  open: boolean;
+  stratumFormInitialValues: IStratumForm
+  onCancel: () => void;
+  onSave: (formikProps: FormikProps<IStratumForm> | null) => void;
+}
+
+const StratumDialog = (props: IStratumDialogProps) => {  
```

Comment:
nit: it would be good to see this in it's own file, it will keep this form page a little cleaner

Reply:
Done!